proto/asgt/modelregistry/v1/model_registry.proto

```diff
@@ -25,11 +25,17 @@ service ModelRegistry {
 }
 
 message RegisterModelRequest {
+  message TargetMetricsUnion {
+    oneof TargetMetricOneOf{
+      asgt.type.TargetMetrics target_metrics_v1 = 1;
+      asgt.type.TargetMetricsV2 target_metrics_v2 = 2;
+    }
+  }
```

Comment:
The idea here isn't bad, but this breaks backwards compatiblity. I think we should just introduce a 9th field here, simply `repeated asgt.type.TargetMetricsV2 target_metrics_v2 = 9;` and in the services we'll just check which ones are being used.

Reply:
Or actually more like `map<string, asgt.type.TargetMetricsV2> per_class_metric = 9;`